starter-code/src/main.js

```diff
@@ -25,23 +25,185 @@ var cards = [
   { name: 'thor',            img: 'thor.jpg' }
 ];
 
+// waiting time when 2 cards are in the front before moving to the next action
+var WAITING_TIME_MS = 1500;
+
+
+class GameDomHandler{
+
+  /**
+   *
+   * @param game: instance of MemoryGame
+   * @param gameDivMapping : array of gameDivMapping as they appear on the dom
+   */
+  constructor(game){
+
+    this.game = game;
+
+    // array of objects with keys (div, card)
+    // the 'div' key returns the dom element relative to a card
+    // the 'card' element retuns the internal representation of a card as in the class MemoryGame
+    // This is a way to keep always together the two objects : the div and the internal representation
+    this.gameDivMapping = [];
+
+    // array of cards currently returned on the dom
+    this.cardsCurrentlyOnPlay = [];
+
+  }
+
+  /**
+   * Renders the game on the DOM
+   */
+  renderGameOnDom(){
+    // renders on the dom
+    let html = '';
+    this.game.cards.forEach(function (pic) {
+
+      // create html
+      html += '<div class="card" data-card-name="'+ pic.name +'">';
+      html += '  <div class="back" name="'+ pic.img +'"></div>';
+      html += '  <div class="front" style="background: url(img/'+ pic.img +') no-repeat"></div>';
+      html += '</div>';
+    });
+    $('#memory_board').html(html);
+
+    // fill in the array gameDivMapping
+    var domCards =  document.querySelectorAll(".card");
+    for (var i = 0; i < domCards.length; i++){
+      this.gameDivMapping.push({
+        "card": this.game.cards[i],
+        "div" : domCards[i],
+        "index": i
+      })
+    }
+  }
+
+  /**
+   * Add event listenners when click on cards
+   * @param e: event to handle
+   */
+  addOnClickListenners(){
+    this.gameDivMapping.forEach(cardELement => {
+      cardELement['div'].addEventListener(
+        "click",
+          e => this.onClickEventListenner(e, cardELement)
+      );
+    })
+  }
+
+  /**
+   * Event listenner on one div
+   * @param e: event to handle
+   * @param cardElement: object witj keys 'div' and 'card'. This should be an element belonging to the array this.gameDivMapping
+   */
+  onClickEventListenner(e, cardElement){
+
+    // Do nothing if two or more cards are returned
+    if (this.cardsCurrentlyOnPlay.length < 2){
```

Comment:
I would also do an early return (like you did below) for this condition. Early returns reduce the need for indentation, making things easier to read.

```suggestion
    if (this.cardsCurrentlyOnPlay.length >= 2) {
      return;
    }
```

Reply:
You could also combine the conditions like this:

```suggestion
    if (this.cardsCurrentlyOnPlay.length >= 2 || (
        this.cardsCurrentlyOnPlay.length === 1 &&
        this .cardsCurrentlyOnPlay[0].index === cardElement.index
    )) {
      return;
    }
```